tests/unit/test_transfer_client.py

```diff
@@ -3,22 +3,56 @@
 from tests.framework import (
     CapturedIOTestCase, get_client_data,
     GO_EP1_ID, GO_EP2_ID, SDKTESTER1A_NATIVE1_RT)
+from globus_sdk.exc import TransferAPIError
 
 
 class TransferClientTests(CapturedIOTestCase):
-    def test_get_endpoint(self):
+
+    def clean(self):
         """
-        Get endpoint on go#ep1 and go#ep2, validate results
+        deletes all endpoints managed by SDK Tester
+        deletes all files and folders in SDK Tester's home directory
+        on both go#ep1 and go#ep2
+        """
+        r = self.tc.endpoint_search(filter_scope="my-endpoints")
+        for ep in r:
+            print "DELETING: ", ep["id"],
+            self.tc.delete_endpoint(ep["id"])
+
+        # TODO: Clean home directories
+
+    def setUp(self):
+        """
+        does an auth flow to create an authorized client
+        calls clean to prevent colisions with pre-exisiting data
+        sets up a test endpoint
         """
         ac = globus_sdk.NativeAppAuthClient(
             client_id=get_client_data()["native_app_client1"]["id"])
         authorizer = globus_sdk.RefreshTokenAuthorizer(
             SDKTESTER1A_NATIVE1_RT, ac)
-        tc = globus_sdk.TransferClient(authorizer=authorizer)
+        self.tc = globus_sdk.TransferClient(authorizer=authorizer)
+
+        self.clean()
+
+        r = self.tc.create_endpoint({"display_name": "SDK Test Endpoint"})
+        self.test_ep_id = r["id"]
```

Comment:
We can roll with this for now, but come to think of it, we should think about what parts of this should be in `setUp` and which should maybe be in `setUpClass`.

I think having endpoint cleanup and creation as part of `setUp` is probably right, and maybe the `TransferClient` should be instantiated in `setUpClass`.